catt/controllers.py

```diff
@@ -447,6 +458,40 @@ def restore(self, data):
                             title=data["title"], thumb=data["thumb"])
 
 
+class DashCastController(CastController):
+    app_ready = "Application ready"
+
+    def __init__(self, cast, name, app_id, prep=None):
+        self._controller = PyChromecastDashCastController()
+        super(DashCastController, self).__init__(cast, name, app_id, prep=prep)
+
+    def load_url(self, url, **kwargs):
+        if self._is_showing_url():
+            # the catch with Chromecast is that it becomes unresponsive after loading a URL
+            # because the new URL does not know it's on Chromecast and therefore does not answer
+            # so the only solution is to kill the DashCast app and load it again
+            self.kill()
+            while self._is_app_loaded():
+                time.sleep(1)
+
+        self._load_url(url)  # this will actually take us to the "Waiting for URL" screen
+        while not self._is_waiting_for_url():
+            time.sleep(1)
+        self._load_url(url)  # this will finally load the URL
```

Comment:
I have found a way to make this work without going through these hoops:
Override ```_prep_app``` in your ```dashcast``` controller, so that it always kills the running app.
Add a ```backdrop_ready``` flag to ```cast_listener```.
In the overridden ```prep_app```, kill the running app and wait for the flag to be set.